cmd/infracost/main.go

```diff
@@ -243,18 +243,22 @@ func loadGlobalFlags(ctx *config.RunContext, cmd *cobra.Command) error {
 }
 
 // saveOutFile saves the output of the command to the file path past in the `--out-file` flag
-func saveOutFile(cmd *cobra.Command, outFile string, b []byte) error {
-	return saveOutFileWithMsg(cmd, outFile, fmt.Sprintf("Output saved to %s\n", outFile), b)
+func saveOutFile(ctx *config.RunContext, cmd *cobra.Command, outFile string, b []byte) error {
+	return saveOutFileWithMsg(ctx, cmd, outFile, fmt.Sprintf("Output saved to %s", outFile), b)
 }
 
 // saveOutFile saves the output of the command to the file path past in the `--out-file` flag
-func saveOutFileWithMsg(cmd *cobra.Command, outFile, successMsg string, b []byte) error {
+func saveOutFileWithMsg(ctx *config.RunContext, cmd *cobra.Command, outFile, successMsg string, b []byte) error {
 	err := ioutil.WriteFile(outFile, b, 0644) // nolint:gosec
 	if err != nil {
 		return errors.Wrap(err, "Unable to save output")
 	}
 
-	cmd.PrintErr(successMsg)
+	if ctx.Config.IsLogging() {
+		log.Info(successMsg)
+	} else {
+		cmd.PrintErrf("%s\n", successMsg)
+	}
```

Comment:
We have this in a few places now, so at some point we need a better way of abstracting the logging vs print to terminal logic, but I didn't want to do that as part of this change.